src/clr-angular/data/datagrid/built-in/filters/datagrid-numeric-filter.ts

```diff
@@ -0,0 +1,154 @@
+import { CustomFilter } from '../../providers/custom-filter';
+import { FiltersProvider, RegisteredFilter } from '../../providers/filters';
+import { DomAdapter } from '../../../../utils/dom-adapter/dom-adapter';
+import { DatagridFilterRegistrar } from '../../utils/datagrid-filter-registrar';
+import { DatagridNumericFilterImpl } from './datagrid-numeric-filter-impl';
+import { ClrCommonStrings } from '../../../../utils/i18n/common-strings.interface';
+
+@Component({
+  selector: 'clr-dg-numeric-filter',
+  providers: [{ provide: CustomFilter, useExisting: DatagridNumericFilter }],
+  template: `
+        <clr-dg-filter [clrDgFilter]="registered" [(clrDgFilterOpen)]="open">
+            <!--
+                Even though this *ngIf looks useless because the filter container already has one,
+                it prevents NgControlStatus and other directives automatically added by Angular
+                on inputs with NgModel from freaking out because of their host binding changing
+                mid-change detection when the input is destroyed.
+            -->
+            <input class="datagrid-numeric-filter-input" #input_low type="number" name="low" [(ngModel)]="low" *ngIf="open"
+                (keyup.enter)="close()" (keyup.escape)="close()" [placeholder]="commonStrings.minValue" 
+                [attr.aria-label]="commonStrings.minValue" />
+                <span class="datagrid-filter-input-spacer"></span>
+            <input class="datagrid-numeric-filter-input" #input_high type="number" name="high" [(ngModel)]="high" *ngIf="open"
+                (keyup.enter)="close()" (keyup.escape)="close()" [placeholder]="commonStrings.maxValue"
+                [attr.aria-label]="commonStrings.maxValue" />
+        </clr-dg-filter>
+    `,
+})
+export class DatagridNumericFilter<T = any> extends DatagridFilterRegistrar<T, DatagridNumericFilterImpl<T>>
+  implements CustomFilter, AfterViewInit {
+  constructor(filters: FiltersProvider<T>, private domAdapter: DomAdapter, public commonStrings: ClrCommonStrings) {
+    super(filters);
+  }
+
+  /**
+   * Customizable filter logic based on high and low values
+   */
+  @Input('clrDgNumericFilter')
+  set customNumericFilter(
+    value: ClrDatagridNumericFilterInterface<T> | RegisteredFilter<T, DatagridNumericFilterImpl<T>>
+  ) {
+    if (value instanceof RegisteredFilter) {
+      this.setFilter(value);
+    } else {
+      this.setFilter(new DatagridNumericFilterImpl(value));
+    }
+  }
+
+  /**
+   * Indicates if the filter dropdown is open
+   */
+  public open: boolean = false;
+
+  /**
+   * We need the actual input element to automatically focus on it
+   */
+  @ViewChild('input_low') public input: ElementRef;
+
+  /**
+   * We grab the ClrDatagridFilter we wrap to register this StringFilter to it.
+   */
+  @ViewChild(ClrDatagridFilter) public filterContainer: ClrDatagridFilter<T>;
+  ngAfterViewInit() {
+    this.filterContainer.openChanged.subscribe((open: boolean) => {
```

Comment:
Should unsubscribe in ngOnDestroy

Reply:
Like this?  https://github.com/vmware/clarity/pull/2996/files#diff-69d97a05a7a4f27670ea86537ac65355R83

Reply:
We're usually using the unsubscribe on the Subscription object
```
this.subscription = this.filterContainer.openChanged.subscribe(...
...
ngOnDestroy() {
  this.subscription.unsubscribe();
}
```
and lately we use array of subscriptions even for single subscriptions, like this:
```
private subscriptions: Subscription[] = [];
....
this.subscriptions.push(this.filterContainer.openChanged.subscribe(...
...
ngOnDestroy() {
  this.subscriptions.forEach(sub => {
    sub.unsubscribe();
  });
}
```
so that it's easy to manage multiple subscriptions if some are added in future.

Reply:
I'll switch it to the array version this evening and also re-run `golden:fix` (why the tests failed).  Thanks!